vyper/ast/folding.py

```diff
@@ -0,0 +1,120 @@
+from typing import (
+    Union,
+)
+
+from vyper.ast import (
+    nodes as vy_ast,
+)
+from vyper.exceptions import (
+    InvalidType,
+)
+
+
+def fold(vyper_ast_node: vy_ast.Module) -> None:
+    """
+    Perform literal folding operations on a Vyper AST.
+
+    Arguments
+    ---------
+    vyper_ast_node : Module
+        Top-level Vyper AST node.
+    """
+    while True:
+        changed_nodes = 0
+        changed_nodes += replace_literal_ops(vyper_ast_node)
+        changed_nodes += replace_subscripts(vyper_ast_node)
+        if not changed_nodes:
+            return
```

Comment:
Can we change this to a conditional `while` loop?
```suggestion
    changed_nodes = 42  # non-zero value to start the loop
    while changed_nodes > 0:
        changed_nodes = 0
        changed_nodes += replace_literal_ops(vyper_ast_node)
        changed_nodes += replace_subscripts(vyper_ast_node)
```

Reply:
Yes, that looks much better